docs/compile.md

```diff
@@ -1,64 +1,106 @@
 
-When publishing contracts that import other contract,  the main contract and all of its imported contracts  will be published - each to their own address.
+When the "Compilation Details" button is clicked (**G. in fig. 1**), a modal opens displaying detailed information about the current selected contract.
 
-**Published data contains the `abi` and the solidity source code.**
+If you just want to get the **ABI** or the **Bytecode** - click the appropriate button see **H. in fig. 1**.
 
-After a contract is published, **a modal will pop up**.  This modal contains the contract's address as well as the addreses of the contracts that it imported and the address of the contract's **metadata**.
+### Passive Contract Verification
+When you publish your metadata to IPFS and deploy your code to the mainnet or a public testnet, the contract verification service **[Sourcify](https://sourcify.dev/)**,  will verifiy your contracts without you needing to do anything.
 
-When the "Compilation Details" button is clicked (**H. in image**), a modal opens displaying detailed information about the current selected contract.
+### Compile and Run script
+The Compile and Run script button (**E. in fig. 1**) is for compiling and then immediately running a script.  It's a time saver so that you can write some code, automatically run script that sets state of the contract - thus allowing you to quickly understand how the code is working.  ([see more](running_js_scripts.html?#compile-a-contract-and-run-a-script-on-the-fly))
 
 ### Compilation Errors and Warning
 Compilation Errors and Warning are displayed below the contract section.
 At each compilation, the static analysis tab builds a report.
 
 It is important to address reported issues even if the compiler doesn't complain. ([see more](static_analysis.html))
 
-### Custom Solidity Compilers
-For those writing your own custom solidity compiler, you can import that by clicking the **+** button (**X. in the image**) to open a modal where you can input the url of the compiler to be loaded.
+## Advanced Compiler Configurations
+Clicking on Advanced Compiler Configurations will open this panel (**M. in fig. 2 below**).
+
+![](images/a-sol-comp-adv.png)
+
+There is a radio button to choose whether to configure the compiler through the interface (**N in fig 2**) or to use a JSON file for the cofiguration (**R in fig 2**). 
+
+### Solidity or YUL
+Since the Solidity version `0.5.7`, it is possible to compile `Yul` files. Please read the ([solidity documentation about Yul](https://solidity.readthedocs.io/en/latest/yul.html)) which contain some code examples.
+You can use the language dropdown (**O. in fig 2**) to switch the language. **This dropdown list is only available for versions greater than or equal to `0.5.7`.**
+
+### Select an Ethereum fork
+The "fork selection" dropdown list (**P. in fin fig 2**) allows to compile code against a specific **ethereum hard fork**.
```

Comment:
remove `fin`

Reply:
the term `fork` has been used throughout this section and on UI, it is EVM, this can be confusing